### PR TITLE
application_starts_on_login: Focus on xterm window before closing it

### PR DESCRIPTION
**Pull request by ggardet**
- Related ticket: https://progress.opensuse.org/issues/66099
- Verification run: https://openqa.opensuse.org/t1460594

### CODE REVIEW

**Comment by Vogtinator** [comment]
This actually sounds like a GNOME bug to me. Something probably steals focus during start and doesn't give it back. This also appears to affect the Wayland session only, which makes it seem even more like a bug.

@DimStar77 suggested to make this a softfail with a match on unfocused xterm, which I agree with. I'm not sure how effective such a needle would be though, considering it's only a slight change in brightness...

**Comment by ggardet** [comment]
> This actually sounds like a GNOME bug to me. Something probably steals focus during start and doesn't give it back. This also appears to affect the Wayland session only, which makes it seem even more like a bug.
> 
> @DimStar77 suggested to make this a softfail with a match on unfocused xterm, which I agree with. I'm not sure how effective such a needle would be though, considering it's only a slight change in brightness...

I really fear to have lots of mismatches if we try to detect it, but I can give it a try.

File: tests/x11/gnomecase/application_starts_on_login.pm [hunk 1]
```diff
@@ -163,7 +163,13 @@ sub run {
     send_key "alt-f4";
 
     handle_relogin;
-    assert_screen 'xterm';
+    assert_screen [qw(xterm xterm-without-focus)];
+    if (match_has_tag 'xterm-without-focus') {
+        record_soft_failure("poo#66099 - Focus not stolen with GNOME 3.36+ https://gitlab.gnome.org/GNOME/mutter/-/issues/1207");
+        # focus it
+        assert_and_click 'xterm-without-focus';
```
Comment (Vogtinator) [inline tests/x11/gnomecase/application_starts_on_login.pm:170]:
```suggestion
        click_lastmatch;
```

Comment (ggardet) [reply tests/x11/gnomecase/application_starts_on_login.pm:170]: I was not aware of `click_lastmatch`. Thanks!

**Comment by ggardet** [comment]
VR passed.

**Comment by Vogtinator** [comment]
LGTM, though I wonder whether the needle might even match on focused xterm. However, I don't see any test run where this could be tested with...

File: tests/x11/gnomecase/application_starts_on_login.pm [hunk 2]
```diff
@@ -163,7 +163,13 @@ sub run {
     send_key "alt-f4";
 
     handle_relogin;
-    assert_screen 'xterm';
+    assert_screen [qw(xterm xterm-without-focus)];
+    if (match_has_tag 'xterm-without-focus') {
+        record_soft_failure("poo#66099 - Focus not stolen with GNOME 3.36+ https://gitlab.gnome.org/GNOME/mutter/-/issues/1207");
```
Comment (DimStar77) [inline tests/x11/gnomecase/application_starts_on_login.pm:168]: *not* stolen?

Comment (ggardet) [reply tests/x11/gnomecase/application_starts_on_login.pm:168]: `xterm` does not steal the focus. I could rephrase: `xterm window does not get the focus` ?

Comment (DimStar77) [reply tests/x11/gnomecase/application_starts_on_login.pm:168]: ah, that's how you see it; I'd say something like 'xterm not focused'

Comment (ggardet) [reply tests/x11/gnomecase/application_starts_on_login.pm]: Ok, updated accordingly.

**Review by DimStar77 — APPROVED** [review]

**Pull request merged**